src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,512 @@
+        String[] words = chatInput.split(SPACING);
+        int taskNumber = Integer.parseInt(words[1]) - accountForArray;
+        return taskNumber;
+    }
+
+    /**
+     * Checks if input string is /at
+     *
+     * @param input input of entry
+     * @return true if entry is /at
+     */
+    private static boolean checkAtEntry(String input) {
+        if (input.equals(ENTRY_AT) || input.equals("(at:")) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Checks if input string is /by
+     *
+     * @param input input of entry
+     * @return true if entry is /by
+     */
+    private static boolean checkByEntry(String input) {
+        if (input.equals(ENTRY_BY) || input.equals("(by:")) {
+            return true;
+        }
+        return false;
+    }
+
+
+    /**
+     * Adds a to-do item to the task list
+     *
+     * @param chatInput input of user
+     */
+    private static void addToDoItem(String chatInput, boolean isFileItem, boolean isDone) {
+        try {
+            ToDo temp = new ToDo(scanDescription(chatInput, false));
+            if (isFileItem) {
+                if (isDone) {
+                    temp.setDone();
+                }
+                tasks.add(temp);
+            } else {
+                tasks.add(temp);
+                System.out.println(MESSAGE_TASK_ADDED + temp);
+                printTaskNumber();
+            }
+            saveFile();
+        } catch (DukeException e) {
+            System.out.println(ERROR_WRONG_TODO_FORMAT);
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println(e);
+        } catch (IOException e) {
+            System.out.println("Save failed! Data might be lost.. :(");
+        }
+    }
+
+    /**
+     * Adds an event item to the task list
+     *
+     * @param chatInput input of user
+     */
+    private static void addEventItem(String chatInput, boolean isFileItem, boolean isDone) {
+        try {
+            Event temp = new Event(scanDescription(chatInput, false), getTimeOfEvent(chatInput, false));
+            if (isFileItem) {
+                if (isDone) {
+                    temp.setDone();
+                }
+                tasks.add(temp);
+            } else {
+                tasks.add(temp);
+                System.out.println(MESSAGE_TASK_ADDED + temp);
+                printTaskNumber();
+            }
+            saveFile();
+        } catch (DukeException e) {
+            System.out.println(ERROR_WRONG_EVENT_FORMAT);
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println(e);
+        } catch (IOException e) {
+            System.out.println("Save failed! Data might be lost.. :(");
+        }
+
+    }
+
+    /**
+     * Adds a deadline item to the task list
+     *
+     * @param chatInput input of user
+     */
+    private static void addDeadlineItem(String chatInput, boolean isFileItem, boolean isDone) {
+        try {
```

Comment:
Consider having the try catch clauses be more specific to which lines of code would throw them,